articles/client-auth/v2/mobile-desktop.md

```diff
@@ -0,0 +1,433 @@
+:::
+
+As an example, your HTML snippet for your authorization URL might look as follows:
+
+```html
+<a href="https://${account.namespace}/authorize?
+  audience=appointments:api&
+  scope=appointments%20contacts&
+  response_type=code&
+  client_id=${account.clientId}&
+  code_challenge=E9Melhoa2OwvFrEMTJguCHaoeK1t8URWbuGJSstw-cM&
+  code_challenge_method=S256&
+  redirect_uri=com.myclientapp://myclientapp.com/callback">
+  Sign In
+</a>
+```
+
+If all goes well, you'll receive an HTTP 302 response:
+
+```text
+HTTP/1.1 302 Found
+Location: https://YOUR_APP/callback?code=AUTHORIZATION_CODE
+```
+
+Note the authorization code included at the end of the included URL.
+
+### Step 3: Obtain an ID Token
+
+Using the authorization code obtained in step 2, you can obtain the ID token by making the appropriate `POST` call to the [tokens endpoint](api/authentication#authorization-code-pkce-).
+
+```har
+{
+  "method": "POST",
+  "url": "https://${account.namespace}/oauth/token",
+  "headers": [
+    { "name": "Content-Type", "value": "application/json" }
+  ],
+  "postData": {
+    "mimeType": "application/json",
+    "text": "{\"grant_type\":\"authorization_code\",\"client_id\": \"${account.clientId}\",\"code_verifier\": \"YOUR_GENERATED_CODE_VERIFIER\",\"code\": \"YOUR_AUTHORIZATION_CODE\",\"redirect_uri\": \"com.myclientapp://myclientapp.com/callback\", }"
+  }
+}
+```
+
+Request Parameters:
+
+* `grant_type`: Set this field to `authorization_code`
+* `client_id`: Your application's Client ID
+* `code_verifier`: The cryptographically random key used to generate the `code_challenge` passed to the `/authorize` endpoint
+* `code`: The authorization code received from the initial `authorize` call
+* `redirect_uri`: the `redirect_uri` passed to `/authorize` (these two values must match exactly)
+
+If all goes well, you'll receive an HTTP 200 response with the following payload:
+
+```json
+{
+  "access_token":"eyJz93a...k4laUWw",
+  "refresh_token":"GEbRxBN...edjnXbL",
+  "id_token":"eyJ0XAi...4faeEoQ",
+  "token_type":"Bearer",
+  "expires_in":86400
+}
+```
+
+:::panel-info Access Tokens
+You can use `access_token` to call the [Authentication API's `/userinfo` endpoint](/api/authentication#get-user-info).
+:::
+
+### Keep the User Logged In
+
+Auth0 assists in authenticating a user, but your application must keep track of whether or not a user is logged in. You can keep a global variable or a singleton object inside your application to do this. You can also use this object to store information about the user (such as name and profile image) so that you can deliver a personalized user experience within your application.
+
+## The `id_token`
```

Comment:
Can you remove this paragraph? Doesn't really fit it. If we wanted to elaborate further we should show how to validate and decode the token, but I don't think it's needed here. Maybe we can iterate on it at some later point.